src/core/ddsi/src/ddsi_security_omg.c

```diff
@@ -415,7 +441,11 @@ static void clear_pending_matches_by_local_guid(dds_security_context *sc, struct
     if (guid_compare(&match->guids.local_guid, local_guid) == 0)
     {
       ddsrt_avl_delete(&pending_match_index_treedef, &index->pending_matches, match);
-      clear_and_free_pending_match(sc, match);
+      if (match->expiry.v != DDS_NEVER)
+        ddsrt_fibheap_delete(&pending_match_expiry_fhdef, &index->expiry_timers, match);
+      inc_guid(&match->guids.remote_guid);
+      next = ddsrt_avl_lookup_succ_eq(&pending_match_index_treedef, &index->pending_matches, &match->guids);
```

Comment:
Doing this instead:
```suggestion
      next = ddsrt_avl_lookup_succ(&pending_match_index_treedef, &index->pending_matches, &match->guids);
```
saves you from having to define ``inc_guid``

Reply:
Thanks, will use your suggestion.